### PR TITLE
🐛 util/version: Add option to compare without prereleases

### PR DESCRIPTION
**Pull request by SubhasmitaSw**
**What this PR does / why we need it**:
/kind bug

Rif #4938 

cc @sbueringer

### CODE REVIEW

File: util/version/version.go [hunk 1]
```diff
@@ -201,12 +201,21 @@ func WithBuildTags() CompareOption {
 	}
 }
 
+func WithoutPreReleases() CompareOption {
+	return func(c *comparer) {
+		c.buildTags = false
```
Comment (sbueringer) [inline util/version/version.go:206]:
```suggestion
		c.withoutPreReleases = true
```
Please add a new field for this flag

File: util/version/version.go [hunk 2]
```diff
@@ -201,12 +201,21 @@ func WithBuildTags() CompareOption {
...
+}
+
 // Compare 2 semver versions.
 // Defaults to doing the standard semver comparison when no options are specified.
 // The comparison logic can be modified by passing additional compare options.
 // Example: using the WithBuildTags() option modifies the compare logic to also
 // consider build tags when comparing versions.
 func Compare(a, b semver.Version, options ...CompareOption) int {
+	a.Pre = nil
+	b.Pre = nil
```
Comment (sbueringer) [inline util/version/version.go:217]:
```suggestion
	if c. withoutPreReleases {
		a.Pre = nil
		b.Pre = nil
	}
```

And let's move it below the code where we collect the options (so ~ l.223)

File: util/version/version.go [hunk 3]
```diff
@@ -201,12 +201,23 @@ func WithBuildTags() CompareOption {
...
+	}
+}
+
 // Compare 2 semver versions.
 // Defaults to doing the standard semver comparison when no options are specified.
 // The comparison logic can be modified by passing additional compare options.
 // Example: using the WithBuildTags() option modifies the compare logic to also
 // consider build tags when comparing versions.
 func Compare(a, b semver.Version, options ...CompareOption) int {
+	if c. withoutPreReleases {
```
Comment (sbueringer) [inline util/version/version.go:216]:
```suggestion
	if c.withoutPreReleases {
```

Comment (sbueringer) [reply util/version/version.go:216]: You have to also add the field to the struct l.177

**Comment by sbueringer** [comment]
/retitle 🐛 util/version: Add option to compare without prereleases

File: util/version/version.go [hunk 4]
```diff
@@ -201,17 +202,29 @@ func WithBuildTags() CompareOption {
...
+}
+
 // Compare 2 semver versions.
 // Defaults to doing the standard semver comparison when no options are specified.
 // The comparison logic can be modified by passing additional compare options.
 // Example: using the WithBuildTags() option modifies the compare logic to also
 // consider build tags when comparing versions.
 func Compare(a, b semver.Version, options ...CompareOption) int {
+
 	c := &comparer{}
```
Comment (sbueringer) [inline util/version/version.go:218]:
```suggestion
	c := &comparer{}
```
@SubhasmitaSw Last nit, golangci-lint doesn't like the leading newline

Comment (SubhasmitaSw) [reply util/version/version.go:218]: @sbueringer  oh okay, might have missed it.

**Comment by sbueringer** [comment]
@SubhasmitaSw Thank you!!
Please squash the commits, then we should be good to go/merge :)

(We're using prow in merge mode to be able to generate release notes, so the commits are preserved after merge)

**Comment by SubhasmitaSw** [comment]
> @SubhasmitaSw Thank you!! Please squash the commits, then we should be good to go/merge :)
> 
> (We're using prow in merge mode to be able to generate release notes, so the commits are preserved after merge)

Thank you @sbueringer for helping through.

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by fabriziopandini** [comment]
/approve cancel
it seems there are lint errrors

**Comment by sbueringer** [comment]
Hm strange not sure how I missed that, maybe it didn't run before. @SubhasmitaSw Can you please add a godoc comment for the func?

**Comment by SubhasmitaSw** [comment]
> Hm strange not sure how I missed that, maybe it didn't run before. @SubhasmitaSw Can you please add a godoc comment for the `WithoutPreReleases` func?

sure, and apology for being late.

**Comment by sbueringer** [comment]
> > Hm strange not sure how I missed that, maybe it didn't run before. @SubhasmitaSw Can you please add a godoc comment for the `WithoutPreReleases` func?
> 
> sure, and apology for being late.

No worries. Good first issues are intentional not time-critical (most of our other issues are also not time-critical)

/lgtm
/approve

**Comment by sbueringer** [comment]
/hold 
for golangci-lint results. The workflows were not approved to run before.

**Comment by sbueringer** [comment]
All green
/hold cancel

**Pull request merged**